clang-tidy-checks/setup.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+function info() {
+  echo "[info]: " $1
+}
+
+function success() {
+  echo "success!"
+}
+
+function check_requirements() {
+  info "checking requirements"
+  cmake --version &&
+  gcc --version &&
+  python3 --version &&
+  ninja --version &&
+  lld --version &&
+  success
+}
+
+function clone_llvm() {
+  info "cloing llvm"
+  git clone https://github.com/llvm/llvm-project.git &&
+  git fetch --all --tags &&
+  git checkout -b tags/llvmorg-11.0.0 &&
+  success
+}
+
+function apply_patches() {
+  info "applying patches"
+  cd llvm-project
+  patch potential-unbounded-loop-check.diff &&
```

Comment:
(see above comment about `set -e`)
```suggestion
  patch potential-unbounded-loop-check.diff
```